src/sync.rs

```diff
@@ -10,7 +10,9 @@ pub(crate) use loom::thread::yield_now;
 
 #[cfg(not(loom))]
 pub(crate) mod atomic {
-    pub(crate) use core::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use core::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicUsize};
+    #[cfg(target_pointer_width = "64")]
```

Comment:
This doesn't seem right. The `#[cfg]` for the `use` of this is gated behind `all(std,not(miri))`, but the gate here is different. I don't think it's obvious what the "right" gate is, but they do need to match. My current thinking is that since `due_time` is stored in nanoseconds, and 2^32 nanoseconds is only ~4 seconds, `due_time` is likely to wrap, and thus be mostly broken, on 32-bit systems. Which suggests that we should disable it entirely on 32-bit systems. So, I think `target_pointer_width = "64"` should be added to the `due_time` `#[cfg]`, and then the documentation should be updated to reflect this limitation on 32-bit systems and `no_std`.

Reply:
What do you think about `#[cfg(all(feature = "std", target_pointer_width = "64", not(miri), not(loom)))]` being the gate for all "time" related things. I have added documentation at the top of domain explaining each attribute of the `#[cfg]`. Its very verbose, but I believe it best describes the constraints.

Reply:
That's great, thanks! I think it should probably be a part of the public docs once I get around to adding those, but this is good for now :+1: